### ISSUE TITLE
Long read CRAM track produces bad layout in certain case

### ISSUE THREAD

**Issue by cmdcolin** [GMOD/jbrowse-components#1654]
This is a random thing I stumbled on using minion cram data

There is something weird going on here where the SNPCoverage track is populated but the PileupTrack is not, and apparently for the CRAM track only. Scrolling around in this area I also saw a populated layout with onhover mouse over shadows, but when clicked, it have an error with feature.toJSON being called on undefined feature

**Comment by cmdcolin** [comment]
There may be some issue with the file


Using samtools

```
samtools view https://s3.amazonaws.com/jbrowse.org/genomes/hg19/NA12878/NA12878-minion-ul_GRCh37.cram 1:38366856-38565648|wc
[main_samview] retrieval of region "1:38366856-38565648" failed due to truncated file or corrupt BAM index file
      1      26 1005617

```

Would maybe be interesting if @gmod/cram could report this type of error but unsure if needed...I can't guarantee this file was created correctly so maybe close for now

**Comment by cmdcolin** [comment]
To be clear this file was created by me at some point in time

**Comment by cmdcolin** [comment]
Tentatively re-opening as I have found a bad layout again

http://s3.amazonaws.com/jbrowse.org/code/jb2/master/index.html?config=test_data%2Fconfig_demo.json&session=share-fVlQIJG-aO&password=iouXH

This file passes samtools quickcheck

samtools quickcheck https://s3.amazonaws.com/jbrowse.org/genomes/hg19/hg002/HG002_ONTrel2_16x_RG_HP10xtrioRTG.cram && echo 'all ok' || echo 'fail'

**Comment by cmdcolin** [comment]
![localhost_3000__config=test_data%2Fconfig_demo json session=local-Cg5fipHMJ (1)](https://user-images.githubusercontent.com/6511937/108422928-6d05fd00-71f4-11eb-9c72-11cc0a99d2e2.png)


Screenshot of this region

we could try again as a first step to verify this file but the behavior seems weird

**Comment by cmdcolin** [comment]
Again, note that SNPCoverage is rendered but the pileup is not

**Comment by cmdcolin** [comment]
igv.js may have issue also  https://tinyurl.com/y72bfsps
it presents a blank screen

![igvjs](https://user-images.githubusercontent.com/6511937/108427347-41861100-71fa-11eb-8336-64685559a66a.png)


igv proper seems ok

![Screenshot from 2021-02-18 15-01-32](https://user-images.githubusercontent.com/6511937/108427322-37fca900-71fa-11eb-8854-e609cea377ea.png)

**Comment by cmdcolin** [comment]
Another link on a different file (first one is)

https://s3.amazonaws.com/jbrowse.org/genomes/hg19/hg002/HG002_ONTrel2_16x_RG_HP10xtrioRTG.cram


this one is

https://s3.amazonaws.com/jbrowse.org/genomes/hg19/ultra-long-ont_hs37d5_phased.cram




http://localhost:3000/?config=test_data%2Fconfig_demo.json&session=share-L1WDPbi0O6&password=Qdjnq

**Comment by cmdcolin** [comment]
Happens found a 3rd third-party file in the same region as the one above, and there are sometimes layout features that go through the blank region



http://s3.amazonaws.com/jbrowse.org/code/jb2/master/index.html?config=test_data%2Fconfig_demo.json&session=share-bWEbjv7z96&password=X6c6h

This link should have two different cram files with the same rendering problem


ROBBBBBB

Note: sometimes the NCBI server that this new file is fetched from erroneously and sporadically returns 200 for a range request so refresh if that happens. This should be unrelated to the issue we are seeing as the s3 bucket shows it too

**Comment by cmdcolin** [comment]
![s3 amazonaws com_jbrowse org_code_jb2_master_index html_config=test_data%2Fconfig_demo json session=local-cUs1F69NG (3)](https://user-images.githubusercontent.com/6511937/108575541-d448ad80-72d7-11eb-9e86-dbc2980b502f.png)

**Comment by cmdcolin** [comment]
possible (probable) fix here :) https://github.com/GMOD/cram-js/pull/84

**Comment by cmdcolin** [comment]
I went ahead and merged https://github.com/GMOD/cram-js/pull/84 (e.g. the linear scan) and released as @gmod/cram 1.5.9 after https://github.com/GMOD/cram-js/pull/85 was found to be flawed in 1.5.8. I put the change into master branch here. I believe it will be solid, if there are concerns we can possibly revisit on @gmod/cram repo...
